authentication/views.py

```diff
@@ -77,3 +93,40 @@ def post(self, request):
             print("Exception", e)
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
+
+class PasswordRecoveryAPIView(APIView):
+    def post(self, request):
+        email = request.data.get('email')
+        try:
+            validate_email(email)
+        except ValidationError:
+            return Response({'error': 'Invalid email address'}, status=400)
+
+        try:
+            user = CustomUser.objects.get(email=email)
+        except CustomUser.DoesNotExist:
+            return Response({'error': 'User with this email does not exist'}, status=404)
+
+        jwt_token = Utils.generate_token(user.email, user.pk)
+        reset_link = f"{settings.FRONTEND_URL}/auth/password-reset/{jwt_token}/"
+        try:
+            Utils.send_password_reset_email(email, reset_link)
+        except Exception as e:
```

Comment:
Catching a general ```Exception``` is usually not recommended because it can catch more than you intend, making debugging harder. It's better to catch specific exceptions related to email sending. Additionally, logging the exception ```e``` can be helpful for debugging purposes

Reply:
Thank you. Removing try-except block, becouse we have it inside that method itself.